22_configuration_management/38_technique_editor.txt

```diff
@@ -0,0 +1,85 @@
+
+You just need to click on a Generic method or drag'n drop it in the area provided for such purpose to add it to the current Technique. Once it's done, you can configure it by clicking on it. Then a new display containing the method details appears instead of the Generic methods list:
+
+image::./images/technique_editor/5-configure-generic-method.png[]
+
+The Generic method details are divided into 3 blocks :
+.Conditions
+Conditions allow user to restrict the execution of the method. 
+.Parameters
+Parameters are in mono or multi line text format. They can contains variables which will be extended at the time of the execution.(syntax:${xxxx})
+.Result classes
+One result class of three will be defined following the execution of a generic method:
+ * One, in case of no modification (Success)
+ * One, in case of modification (Repaired)
+ * One, in case of error (Error)
+
+Theses classes can be used in another Generic methods conditions.
+
+
+==== Create your first Technique
+
+Now we are going to see how to create a simple technique to configure a ntp server, step by step.
+
+===== 1. General information
+Let's start from the beginning. Click on the "New Technique" button and start filling in the General information fields (only name is required).
+In our case:
+* Name: _Configure NTP_
+* Description: _Install, configure and ensure the ntpd is running. Uses a template file to configuration._
+
+===== 2. Add and configure generic methods
+Now, we have to find and add the generic methods which correspond to the actions we want to execute. In our case, we want to add the following methods:
+* Package install (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the package to install. So here, fill in the *package_name* field with the value _ntp_.
+
+* File from template (You can find it in the *File category*)
+  - This method take two parameters. The first one corresponds to the absolute path of the source file containing a template to be expanded. We are going to use a Rudder variable here to get the correct path. Fill in the *source_template* field with the value _${path_technique}/templates/ntp.conf_.
+	The second corresponds to the absolute path of the destination file. Fill in with the value _/etc/ntp.conf_.
+
+* Service restart (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the service we want to restart. So here, fill in the *service_name* field with the value _ntp_.
+* Service ensure running (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the service we want to check. Again, here, fill in the *service_name* field with the value _ntp_.
+
+===== 3. Done!
+And… It’s already done. Rather fast, right? Don’t forget to save you technique. Now you can see it in the *Active Techniques tree*, and  use that it to create a _Directive_ that will be applied on your _Nodes_ thanks to a _Rule_.
```

Comment:
Would also speak of Directive tree, so people go to Directive page and not technique one